File: AiSD5_C2/exercise_2/Python/main.py
def read_year():
    year = int(input('Wprowadz rok do sprawdzenia: '))
    return year

def is_leap_year(year):
    if year % 400 == 0:
        return True
    elif year % 4 == 0:
        if year % 100 != 0:
            return True
        else:
            return False
    else:
        return False

def display_result(year, is_leap):
    if is_leap:
        print(f'Rok {year} jest rokiem przestepnym.')
    else:
        print(f'Rok {year} nie jest rokiem przestepnym.')

def process_leap_year_check():
    year = read_year()
    is_leap = is_leap_year(year)
    display_result(year, is_leap)

if __name__ == "__main__":
    process_leap_year_check()
    input('Wcisnij Enter aby wyjść z programu...')

File: AiSD5_C2/exercise_2/Pascal/main.pas
program LeapYearCheck;

uses 
  CRT, math, sysUtils;

// Function to read the year from user input
function ReadYear: integer;
begin
  write('Wprowadz rok do sprawdzenia: ');
  readln(ReadYear);
end;

// Function to check if a year is leap following the exact flowchart logic
function IsLeapYear(year: integer): boolean;
begin
  // First check: if year mod 400 = 0
  if (year mod 400 = 0) then
    IsLeapYear := true
  else
  begin
    // Second check: if year mod 4 = 0
    if (year mod 4 = 0) then
    begin
      // Third check: if year mod 100 <> 0
      if (year mod 100 <> 0) then
        IsLeapYear := true
      else
        IsLeapYear := false;
    end
    else
      IsLeapYear := false;
  end;
end;

// Procedure to display the result
procedure DisplayResult(year: integer; isLeap: boolean);
begin
  write('Rok ', year);
  if isLeap then
    writeln(' jest rokiem przestepnym.')
  else
    writeln(' nie jest rokiem przestepnym.');
end;

// Main procedure that coordinates all operations
procedure ProcessLeapYearCheck;
var
  year: integer;
  isLeap: boolean;
begin
  // Clear the screen for better visibility
  ClrScr;
  
  // Get the year from user ("Podaj Rok")
  year := ReadYear;
  
  // Check if it's a leap year following flowchart logic
  isLeap := IsLeapYear(year);
  
  // Display the result
  DisplayResult(year, isLeap);
  
  // Wait for user input before closing
  writeln;
  writeln('Wcisnij Enter aby wyjśc z programu...');
  readln;
end;

// Main program
begin
  ProcessLeapYearCheck;
end.